### PR TITLE
Feature: patch update support tun config

### PR DESCRIPTION
**Pull request by littoy**

### CODE REVIEW

File: hub/route/configs.go [hunk 1]
```diff
@@ -92,6 +93,19 @@ func patchConfigs(w http.ResponseWriter, r *http.Request) {
...
+		tunConfig := *&general.Tun
+		if tunConfig.Enable && executor.GetDefaultInterfaceName() == "" {
+			autoDetectInterfaceName, err := commons.GetAutoDetectInterface()
+			if err != nil || autoDetectInterfaceName == "" {
+				return
+			}
+			executor.UpdateInterface(autoDetectInterfaceName)
+		}
+		P.ReCreateTun(tunConfig, P.GetTunLastAddressPrefix(), tcpIn, udpIn)
+		P.ReCreateTun(tunConfig, P.GetTunLastAddressPrefix(), tcpIn, udpIn)
```
Comment (yaling888) [inline hub/route/configs.go]: duplicate

File: hub/executor/executor.go [hunk 2]
```diff
@@ -68,6 +68,21 @@ func ParseWithBytes(buf []byte) (*config.Config, error) {
 	return config.Parse(buf)
 }
 
+func GetDefaultInterfaceName() string {
```
Comment (yaling888) [inline hub/executor/executor.go]: `dialer.DefaultInterface` need to synchronize?

File: listener/listener.go [hunk 3]
```diff
@@ -30,8 +30,9 @@ import (
 )
 
 var (
-	allowLan    = false
-	bindAddress = "*"
+	allowLan          = false
+	bindAddress       = "*"
+	lastAddressPrefix *netip.Prefix
```
Comment (yaling888) [inline listener/listener.go]: `lastAddressPrefix` using in where?

File: constant/metadata.go [hunk 4]
```diff
@@ -72,22 +73,26 @@ func (t Type) MarshalJSON() ([]byte, error) {
...
+	ProcessPath string     `json:"processPath"`
+	UserAgent   string     `json:"userAgent"`
 }
 
 func (m *Metadata) RemoteAddress() string {
-	if m.DstIP != nil {
+	return net.JoinHostPort(m.String(), m.DstPort)
+}
+
+func (m *Metadata) RemoteIPAddress() string {
```
Comment (yaling888) [inline constant/metadata.go]: what's the difference between `func (m *Metadata) RemoteIPAddress() string {` and `func (m *Metadata) Pure() *Metadata {`

**Review by yaling888 — CHANGES_REQUESTED** [review]
cc

File: hub/route/configs.go (hunk 1 above)
Comment (littoy) [reply hub/route/configs.go]: I'll check

File: listener/listener.go (hunk 3 above)
Comment (littoy) [reply listener/listener.go]: this use in P.ReCreateTun . I've not found other method get this param more easy.

File: constant/metadata.go (hunk 4 above)
Comment (littoy) [reply constant/metadata.go]: string vs. *Metatata ?

File: listener/listener.go (hunk 3 above)
Comment (yaling888) [reply listener/listener.go]: but no read it

File: constant/metadata.go (hunk 4 above)
Comment (yaling888) [reply constant/metadata.go]: before calls function `DialContext()` the function `Pure()` has been called.
https://github.com/yaling888/clash/blob/d763900b14d7e0a6dc3933e6fceeebb84ff1b136/tunnel/tunnel.go#L323

Comment (yaling888) [reply constant/metadata.go]: if you disabled Clash dns, maybe do this is better.
```suggestion
func (m *Metadata) Pure() *Metadata {
	- if m.DNSMode == DNSMapping && m.DstIP.IsValid() {
	+ if m.DNSMode != DNSFakeIP && m.DstIP.IsValid() {
	....
```

File: listener/listener.go (hunk 3 above)
Comment (littoy) [reply listener/listener.go]: read by GetTunLastAddressPrefix()

File: constant/metadata.go (hunk 4 above)
Comment (littoy) [reply constant/metadata.go]: let me have some think about dns mode. The problem I'm having is on Direct, a domian  resolve to multi ip, if I want ssh connect one of the ip, may be resolved to an other one. problem like ssh ->  1.1.1.1:22 -> dns map(xxx.com)->2.2.2.2:22

Comment (littoy) [reply constant/metadata.go]: looks like not need RemoteIPAddress any more. I'll revert it.

**Review by yaling888 — APPROVED** [review]

**Pull request merged**